pkg/collector/corechecks/cluster/orchestrator/terminated_resource_bundle.go

```diff
@@ -51,6 +52,12 @@ func (tb *TerminatedResourceBundle) Add(k8sCollector collectors.K8sCollector, re
 		tb.terminatedResources[k8sCollector] = []interface{}{}
 	}
 
+	resource, ok := checkResourceType(k8sCollector, resource)
```

Comment:
Instead of adding `ResourceType` to all structs, can't we check if it's `cache.DeletedFinalStateUnknown` here directly and return? i.e.

```golang
        // Skip objects without known final state
        if _, ok := resource.(cache.DeletedFinalStateUnknown){
               return
               log.Debug("...")
}
```

Thoughts on it @xlucas?

Reply:
no big opinion, sounds fine by me if we want to be super safe in case informers code evolve but not sure it's going to be the case

when we'll work on refactoring the check we can see if that part can be improved with generics